hazelcast/src/main/java/com/hazelcast/nio/tcp/WriteHandler.java

```diff
@@ -172,7 +172,7 @@ private void unschedule() {
         // So the outputBuffer is empty, so we are going to unschedule ourselves.
         scheduled.set(false);
 
-        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty()) {
+        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty() && currentPacket == null) {
```

Comment:
@pveentjer All I wanted to make sure is that if we are confident on the new changes. If we are not, it is not too late to revert. If we think this is much more readable and easily we can easily fix the bugs, then I am OK with the changes and we can leave as is.